fenix/app/src/test/java/org/mozilla/fenix/search/SearchFragmentStoreTest.kt

```diff
@@ -351,6 +357,46 @@ class SearchFragmentStoreTest {
         assertFalse(store.state.showSearchShortcuts)
     }
 
+    @Test
+    fun `GIVEN sponsored suggestions are enabled WHEN the default search engine is selected THEN sponsored suggestions are displayed`() = runTest {
+        val initialState = emptyDefaultState(showSponsoredSuggestions = false, showNonSponsoredSuggestions = false)
+        val store = SearchFragmentStore(initialState)
+        every { settings.showSponsoredSuggestions } returns true
+        every { settings.showNonSponsoredSuggestions } returns false
+
+        store.dispatch(
+            SearchFragmentAction.SearchDefaultEngineSelected(
+                engine = searchEngine,
+                browsingMode = BrowsingMode.Normal,
+                settings = settings,
+            ),
+        ).join()
+
+        assertNotSame(initialState, store.state)
+        assertTrue(store.state.showSponsoredSuggestions)
+        assertFalse(store.state.showNonSponsoredSuggestions)
+    }
+
+    @Test
+    fun `GIVEN non-sponsored suggestions are enabled WHEN the default search engine is selected THEN non-sponsored suggestions are displayed`() = runTest {
+        val initialState = emptyDefaultState(showSponsoredSuggestions = false, showNonSponsoredSuggestions = false)
+        val store = SearchFragmentStore(initialState)
+        every { settings.showSponsoredSuggestions } returns false
+        every { settings.showNonSponsoredSuggestions } returns true
+
+        store.dispatch(
+            SearchFragmentAction.SearchDefaultEngineSelected(
+                engine = searchEngine,
+                browsingMode = BrowsingMode.Normal,
+                settings = settings,
+            ),
+        ).join()
+
+        assertNotSame(initialState, store.state)
+        assertFalse(store.state.showSponsoredSuggestions)
+        assertTrue(store.state.showNonSponsoredSuggestions)
+    }
```

Comment:
I think it could be useful to have a test in the negative here as well, something like: 

`GIVEN sponsored and non-sponsored suggestions are enabled WHEN the non-default search engine is selected THEN neither are displayed`

With this we could probably avoid adding comments as I suggested above.

Reply:
Done and done—I added negative tests to cover all the non-default engines, and a sentence to the `SearchFragmentStore` docs confirming that they're never shown when a non-default engine is selected.